src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -6916,11 +6916,11 @@ int RGWRados::block_while_resharding(RGWRados::BucketShard *bs,
     if (!entry.resharding_in_progress()) {
       ret = fetch_new_bucket_info("get_bucket_resharding_succeeded");
       if (ret < 0) {
-	ldpp_dout(dpp, 0) << "ERROR: " << __func__ <<
-	  " failed to refresh bucket info after reshard when get bucket "
-	  "resharding succeeded, error: " << cpp_strerror(-ret) << dendl;
-	return ret;
+        ldpp_dout(dpp, 0) << "ERROR: " << __func__ <<
+          " failed to refresh bucket info after reshard when get bucket "
+          "resharding succeeded, error: " << cpp_strerror(-ret) << dendl;
       }
+      return ret;
```

Comment:
can you help me understand the reasoning here? this reverts a recent change from https://github.com/ceph/ceph/pull/48899

Reply:
> can you help me understand the reasoning here? this reverts a recent change from #48899

I might not understand entirely, as far as I can see, there are two cases running to this place. Firstly, reshard success and the old shard has already been deleted, so stay at initial NOT_RESHARDING state. Secondly, reshard failed and the shard has changed to  NOT_RESHARDING state by clear_resharding(). No matter which case, it is better to return immediately and continue execute. It is not necessary to wait once, have I understood correctly?

Reply:
thanks @liangmingyuanneo, i think you're right. if `fetch_new_bucket_info()` succeeds, we don't know whether that new bucket instance is resharding, so it's best to return directly to `guard_reshard()` to retry the operation. if we find that it is still resharding, we'll call back into `block_while_resharding()`